app/cdap/components/WranglerGrid/SelectColumnPanel/DataTable/index.tsx

```diff
@@ -0,0 +1,182 @@
+
+const SelectColumnTableContainer = styled(TableContainer)`
+  height: 100%;
+`;
+
+const SelectColumnTable = styled(Table)`
+  display: flex;
+  flex-direction: column;
+`;
+
+const SelectColumnTableHead = styled(TableHead)`
+  height: 54px;
+`;
+
+const SelectColumnTableRow = styled(TableRow)`
+  font-weight: 400;
+  font-size: 16px;
+  line-height: 150%;
+  letter-spacing: 0.15px;
+  color: ${grey[700]};
+  display: grid;
+  grid-template-columns: 8% 45% 45%;
+  align-items: center;
+  height: 100%;
+`;
+
+const SelectColumnTableHeadCell = styled(TableCell)`
+  &.MuiTableCell-head {
+    padding: 0;
+    font-weight: 400;
+    font-size: 16px;
+    line-height: 150%;
+    letter-spacing: 0.15px;
+    color: ${grey[900]};
+    border-bottom: none !important;
+  }
+`;
+
+const FlexWrapper = styled(Box)`
+  display: flex;
+  height: 100%;
+  align-items: center;
+`;
+
+const CenterAlignBox = styled(Box)`
+  text-align: center;
+`;
+
+export default function({
+  columns,
+  transformationDataType,
+  onSingleSelection,
+  selectedColumns,
+  dataQualityValue,
+  isSingleSelection,
+  handleDisableCheckbox,
+  onMultipleSelection,
+  setSelectedColumns,
+  transformationName,
+}: IColumnTableProps) {
+  const handleChange = () => {
+    if (
+      MULTI_SELECTION_COLUMN?.filter(
+        (option) => option.value === transformationName && option.isMoreThanTwo === false
+      ).length > 0
+    ) {
+      if (selectedColumns.length) {
+        setSelectedColumns([]);
+      } else {
+        if (columns?.length > 2) {
+          setSelectedColumns(columns.slice(0, 2));
+        } else {
+          setSelectedColumns(columns);
+        }
+      }
+    } else {
+      if (selectedColumns?.length) {
+        setSelectedColumns([]);
+      } else {
+        setSelectedColumns(columns);
+      }
+    }
+  };
+
+  const columnsToDisplay = transformationDataType?.includes('all')
+    ? columns
+    : columns.filter((eachColumn) =>
+        transformationDataType?.includes(eachColumn?.type[0]?.toLowerCase())
+      );
+
+  return (
+    <SelectColumnTableContainer data-testid="column-table-parent">
+      {columnsToDisplay.length === 0 ? (
+        <FlexWrapper>
+          <CenterAlignBox>
+            {NoDataSVG}
+            <SubHeadBoldFont component="p" data-testid="no-column-title">
+              {T.translate(`${SELECT_COLUMN_LIST_PREFIX}.noColumns`)}
+            </SubHeadBoldFont>
+            <NormalFont component="p" data-testid="no-column-subTitle">
+              {T.translate(`${SELECT_COLUMN_LIST_PREFIX}.noMatchColumnDatatype`)}
+            </NormalFont>
+          </CenterAlignBox>
+        </FlexWrapper>
+      ) : (
+        <SelectColumnTable aria-label="recipe steps table">
+          <Divider />
+          <SelectColumnTableHead>
+            <SelectColumnTableRow>
+              <SelectColumnTableHeadCell>
+                {MULTI_SELECTION_COLUMN?.filter((option) => option.value === transformationName)
```

Comment:
can we use some instead of a filter here?
plz optimize performance

Reply:
Done